### DISCUSSION TITLE
Metadata elements for chant fragments

### DISCUSSION THREAD

**Discussion by mfeustle** [music-encoding/music-encoding#1042] [Ideas]
The immediate application I envision for the features proposed below would be their potential to link separated chant fragments, but broader applications seem possible. 

Texts in chant fragments are generally standardized enough to be recognizable, as a result of their specific liturgical function. It is therefore possible to determine what text would precede or follow what is present on a fragment with a great deal of confidence. 

The current options under the otherChar element, "follows" and "precedes," provide half of what is needed for automating the match of potential adjacent pages.

I'd like to propose the addition of four items, either as elements, or as attributes of the existing otherChar element:

begins-with
ends-with
follows-from
followed-by

These would identify a match when "begins with" matches "followed-by" or when "ends-with" matches "follows from."

For example, suppose we had a fragment from Vespers that cuts off in the middle of the singing of the Magnificat. Since the Latin text is standardized, we know that the words that conclude our hypothetical page, "Fecit potentiam in bracchio suo," will be followed on the next page by "Dispersit superbos mente cordis sui." 

Therefore, our page would have an "ends-with" of "Fecit potentiam in bracchio suo" and a "followed-by" that is "Dispersit superbos mente cordis sui."

Somewhere out there, there is a page that begins with "Dispersit superbos mente cordis sui," which follows from "Fecit potentiam in bracchio suo."

If we structure this data, we can isolate potential matches when "ends-with" matches "follows-from" or when "begins-with" matches "followed-by."

One variable that could be addressed by common practice is the length of these text samples. In the example above, five words suffice to set up a virtual "handshake" between disconnected, once-adjacent pages. 

I appreciate the consideration of this new feature request, and I hope it is useful in advancing discoverability of chant fragments.

**Comment by ahankinson** [comment]
Could you please sketch out some sample XML of how you would see this working? I'm trying to figure out if this can be covered by the existing attributes (you mention `follows` and `precedes`, but not `next` and `prev` which provide similar capability: `follows` <->`prev` and `next` <-> `precedes`. So providing a concrete example would help.

**Reply by mfeustle** [reply]
Thanks. Here's what I'm thinking for our two hypothetical fragments from above, where the focus is on uniting consecutive pages. (Editing with a space after the < to fix Github taking the code too literally!)

Imaginary document 1:

```< otherChar follows-from="magnificat anima mea dominum et" />

< otherChar begins-with="exultavit spiritus meus in Deo" />

< otherChar ends-with="fecit potentiam in bracchio suo" />

< otherChar followed-by="dispersit superbos mente cordis sui" />
```

Imaginary document 2:

```< otherChar follows-from="fecit potentiam in bracchio suo" />

< otherChar begins-with="dispersit superbos mente cordis su" />

< otherChar ends-with="lorem ipsum dolor sit amet" />

< otherChar followed-by="consectetur adipiscing elit sed do" />
```

**Reply by ahankinson** [reply]
Thanks! That's helpful. 

BTW, If you want to put code in github, you can surround it in triple-backticks, which will look like this:

```
<element>Foo</element>
```

I must admit, I'm still not entirely clear on what you are aiming to accomplish. Are you suggesting that, by encoding in this way, you could somehow link two documents using their textual content? Or simply that you could complete a missing part of a fragment with a supplied text? 

I see a few methodological difficulties with your proposal as it is. Fragments do not "standardize" themselves -- it's unlikely that there is a large corpus of fragments that cut off at the same point! In that case, it seems to me that the two most important things to capture for fragment identification are: 

 1. The text on the page, as it is
 2. The full text of the piece, or a textual incipit to identify a standardized text (e.g., "Magnificat anima mea Dominum")

With those two pieces of information you can identify what is on the fragment, and the fragment's missing bits. (This is putting aside the fact that actual MSS contain all sorts of variants, like different spelling, abbreviation, or outright missing words that would also make it difficult to know how to align the two). 

We generally do not do linking between "real" text because it is extremely fragile for computers -- a single space missing, or punctuation, or capitalization, and the link breaks. You would typically link between documents by using XML IDs, since they provide pointers between documents. For example, if we take your example from above, it might be something like:

```
In frag1.mei:

<otherChar xml:id="s1" follows-from="frag2.mei#d1">magnificat anima mea dominum et</otherChar>

In frag2.mei:

<otherChar xml:id="d1" begins-with="frag1.mei#s1">exultavit spiritus meus in Deo</otherChar>
```

The `xml:id` value provides a unique identifier for that element and its text content, while the `frag2.mei#d1` value is a standard way of linking between files -- the filename, (or even a full URL) a hash, and then the identifier of the element you are wanting to link.

**Comment by musicEnfanthen** [comment]
Thank you @mfeustle for bringing this up. Pinging different interest groups that may be interested in the discussion: @music-encoding/ig-metadata @music-encoding/ig-neumes @music-encoding/ig-mensural

**Comment by pe-ro** [comment]
I'm in agreement with @ahankinson.

I'd like to add that using a generic, catch-all kind of element like `<otherChar>` for a very specific purpose such as this is problematic.  I think `<incipText>` and `<explicit>` would be better suited to this task.  They capture the appropriate text and have the necessary attributes (`@prev` and `@next`) for capturing the links between the current document and another.
